nearcache/config_test.go

```diff
@@ -0,0 +1,305 @@
+	config, ncs := configWithNearCacheNames("*", "com.hazelcast.*")
+	assert.Equal(t, ncs[1], assertTrueGetNearCacheConfig(t, config, "com.hazelcast.myNearCache"))
+}
+
+func TestDefaultConfig(t *testing.T) {
+	ncc := nearcache.Config{}
+	if err := ncc.Validate(); err != nil {
+		t.Fatal(err)
+	}
+	target := nearcache.Config{
+		Name:           "default",
+		EvictionConfig: nearcache.EvictionConfig{},
+		PreloaderConfig: nearcache.PreloaderConfig{
+			Enabled:                  false,
+			Directory:                "",
+			StoreIntervalSeconds:     600,
+			StoreInitialDelaySeconds: 600,
+		},
+		InMemoryFormat:    nearcache.InMemoryFormatBinary,
+		SerializeKeys:     false,
+		TimeToLiveSeconds: math.MaxInt32,
+		MaxIdleSeconds:    math.MaxInt32,
+	}
+	assert.Equal(t, target, ncc)
+}
+
+type testCase struct {
+	name string
+	cfg  nearcache.Config
+}
+
+func (tc testCase) Run(t *testing.T) {
+	t.Run(tc.name, func(t *testing.T) {
+		err := tc.cfg.Validate()
+		if !errors.Is(err, hzerrors.ErrInvalidConfiguration) {
+			t.Fatalf("%s: expected ErrInvalidConfiguration", tc.name)
+		}
+	})
+}
+
+func TestConfigInvalid(t *testing.T) {
+	testCases := []testCase{
+		{
+			name: "negative time to live",
+			cfg:  nearcache.Config{TimeToLiveSeconds: -1},
+		},
+		{
+			name: "negative max idle",
+			cfg:  nearcache.Config{MaxIdleSeconds: -1},
+		},
+	}
+	for _, tc := range testCases {
+		tc.Run(t)
+	}
+}
+
+type comparator struct{}
+
+func (c comparator) Compare(a, b nearcache.EvictableEntryView) int {
+	return 0
+}
+
+func TestEvictionConfigInvalid(t *testing.T) {
+	// has both policy and comparator
+	ec1 := nearcache.EvictionConfig{}
+	ec1.SetEvictionPolicy(nearcache.EvictionPolicyNone)
+	ec1.SetComparator(&comparator{})
+	// invalid policy
+	ec2 := nearcache.EvictionConfig{}
+	ec2.SetEvictionPolicy(1000)
+	// size out of range
+	ec3 := nearcache.EvictionConfig{}
+	ec3.SetSize(-1)
+	testCases := []testCase{
+		{
+			name: "has both policy and comparator",
+			cfg:  nearcache.Config{EvictionConfig: ec1},
+		},
+		{
+			name: "invalid policy",
+			cfg:  nearcache.Config{EvictionConfig: ec2},
+		},
+		{
+			name: "size out of range",
+			cfg:  nearcache.Config{EvictionConfig: ec3},
+		},
+	}
+	for _, tc := range testCases {
+		tc.Run(t)
+	}
+}
+
+func TestPreloaderConfigInvalid(t *testing.T) {
+	testCases := []testCase{
+		{
+			name: "negative store initial delay seconds",
+			cfg: nearcache.Config{
+				PreloaderConfig: nearcache.PreloaderConfig{
+					StoreInitialDelaySeconds: -1,
+				},
+			},
+		},
+		{
+			name: "negative store interval seconds",
+			cfg: nearcache.Config{
+				PreloaderConfig: nearcache.PreloaderConfig{
+					StoreIntervalSeconds: -1,
+				},
+			},
+		},
+		{
+			name: "blank directory when enabled",
+			cfg: nearcache.Config{
+				PreloaderConfig: nearcache.PreloaderConfig{
+					Enabled: true,
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		tc.Run(t)
+	}
+}
+
+func TestConfigInvalidNon32bit(t *testing.T) {
```

Comment:
Does the call `int32(mathMaxInt32 + 1)` throw in `check.NonNegativeInt32` in this test?